src/shared/components/Header/HeaderMenuItems.tsx

```diff
@@ -1,68 +1,85 @@
-// No check because typing of asc-ui is not correct.
-// @ts-nocheck
-// this is the reason why the menu items are located in this separate file
-// which is an effort to keep unchecked code to a minimum
-//
-// a block-scoped ignore in Header.tsx would be a more elegant solution
-// however TS does not offer a block-scoped ignore at the time of writing
-// see https://github.com/Microsoft/TypeScript/issues/19573
+import { useMemo } from 'react'
+import { Icon, MenuButton, MenuItem, themeColor } from '@amsterdam/asc-ui'
+import { Link, matchPath, useLocation } from 'react-router-dom'
+import { ExternalLink } from '@amsterdam/asc-assets'
+import styled from 'styled-components'
+import {
+  MenuOrCardItemData as MenuItemData,
+  menuOrCardItems as menuItems,
+} from '../../utils/menuOrCardItems'
+import { getGeneratedPath } from '../../utils/path'
 
-import { MenuButton, MenuItem } from '@amsterdam/asc-ui'
-import { generatePath, Link, matchPath, useLocation } from 'react-router-dom'
-import { getPathTo } from '../../../routes'
+function mapMenuItems(item: MenuItemData) {
+  const itemWithPath: MenuItemData = {
+    title: item.titleShort ?? item.title,
+    path: item.path ?? '',
```

Comment:
According to your `MenuItemData` type, `item.path` can't be null right? So why do the check here? If you are unsure if the parameter item contains the path property it should have been tested before it became a `MenuItemData`?